ext/curl/interface.c

```diff
@@ -1634,7 +1634,10 @@ static bool php_curl_set_callable_handler(zend_fcall_info_cache *const handler_f
 #define HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER(curl_ptr, constant_no_function, handler_type) \
 	case constant_no_function##FUNCTION: { \
 		bool result = php_curl_set_callable_handler(&curl_ptr->handlers.handler_type->fcc, zvalue, is_array_config, #constant_no_function "FUNCTION"); \
-		if (!result) { \
+		if (!result \
+		    || (curl_ptr->handlers.read && !curl_ptr->handlers.read->fcc.function_handler) \
+		    || (curl_ptr->handlers.write && !curl_ptr->handlers.write->fcc.function_handler) \
+		    || (curl_ptr->handlers.write_header && !curl_ptr->handlers.write_header->fcc.function_handler)) { \
```

Comment:
This looks weird.
You probably have to check just ``if (!result)`` or
``if (!result || !curl_ptr->handlers.handler_type->fcc.function_handler)``.

Reply:
yes I just did locally I realised later..